master/custom/release_dashboard.py

```diff
@@ -1,122 +1,730 @@
+            if worker["connected_to"]:
+                for cnf in worker["configured_on"]:
+                    if cnf["builderid"] == self["builderid"]:
+                        yield worker
+
+class Worker(DashboardObject):
+    pass  # The JSON is fine! :)
+
+@total_ordering
+class _BranchTierBase(DashboardObject):
+    """Base class for Branch and Tag"""
+    # Branches have several kinds of names:
+    # 'tag': '3.x' (used as key)
+    # 'version': '3.14'
+    # 'branch': 'main'
+    # To prevent confusion, there's no 'name'
+
+    @cached_property
+    def tag(self):
+        return self["tag"]
+
+    def __hash__(self):
+        return hash(self.tag)
+
+    def __eq__(self, other):
+        if isinstance(other, str):
+            return self.tag == other
+        return self.sort_key == other.sort_key
 
-            failed_builds = failed_builds_by_tier.setdefault(tier, [])
-            failed_builds.append((builder, last_build))
-
-        def tier_sort_key(item):
-            tier, data = item
-            if tier == 'no tier':
-                return 'zzz'  # sort last
-            return tier
-
-        failed_builders = []
-        for branch, failed_builds_by_tier in failed_builds_by_branch_and_tier.items():
-            failed_builders.append((
-                branch,
-                sorted(failed_builds_by_tier.items(), key=tier_sort_key)
-            ))
-
-        def branch_sort_key(item):
-            branch, *_ = item
-            minor = branch.split('.')[-1]
+    def __lt__(self, other):
+        return self.sort_key < other.sort_key
+
+    def __str__(self):
+        return self.tag
+
+@total_ordering
+class Branch(_BranchTierBase):
+    @cached_property
+    def sort_key(self):
+        if self.tag.startswith("3."):
             try:
-                return int(minor)
+                return (1, int(self.tag[2:]))
             except ValueError:
-                return 99
+                return (2, 99)
+        return (0, 0)
 
-        failed_builders.sort(reverse=True, key=branch_sort_key)
+    @cached_property
+    def title(self):
+        if self.tag == '3.x':
+            return 'main'
+        return self.tag
 
-        generated_at = datetime.datetime.now(tz=datetime.timezone.utc)
+    @cached_sorted_property()
+    def problems(self):
+        problems = []
+        for builder in self._root.builders:
+            if builder.branch == self:
+                problems.extend(builder.problems)
+        return problems
 
-        return render_template(
-            "releasedashboard.html",
-            failed_builders=failed_builders,
-            generated_at=generated_at,
-            disconnected_workers=sorted(disconnected_workers.items()),
+    @cached_property
+    def featured_problem(self):
+       try:
+        try:
+            return self.problems[0]
+        except IndexError:
+            return NoProblem()
+       except AttributeError:
+           raise SystemError
```

Comment:
Why `SystemError`? Shouldn't better be something concrete like`BuilderError` or something akin to that?